python/lsst/ts/m1m3/utils/thermocouples.py

```diff
@@ -52,318 +56,758 @@ async def get_scanner_data(
+            "MTCOW12M": -0.02630977616099628,
+            "MTCOW12F": -0.06358752765974884,
+            "MTC038B1": 0.08579040118004144,
+            "MTC038F": 0.05191260458309088,
+            "MTC040B2": 0.09525678437231555,
+            "MTC041B": 0.05055711548803821,
+            "MTC006B": 0.028779326690024476,
+            "MTC006M": 0.0033348720730526082,
+            "MTC006F": -0.010942904009302978,
+            "MTC005B": 0.09879035964222638,
+            "MTCOW1B": 0.19741236383330651,
+            "MTCOW1M": 0.12783461691219244,
+            "MTCOW1F": 0.12997921852216443,
+            "MTC008B1": 0.12564582839222638,
+            "MTC008M": 0.10631251879478754,
+            "MTC008F": 0.07534586603057214,
+            "MTCIW1B": 0.11469022977297705,
+            "MTCIW1F": 0.09993445093047448,
+            "MTC001B": 0.09057911781415662,
+            "MTC001M": 0.09715692746584814,
+            "MTC001F": 0.08035700070803564,
+            "MTC002B": 0.07372366601836511,
+            "MTC002F": 0.06742348155655975,
+            "MTC003B": 0.024823591589383298,
+            "MTC003F": -0.0048543292819278605,
+            "MTC007B1": 0.0107459811496593,
+            "MTC007F": 0.006834750850451599,
+            "MTC004B": 0.09557912629126086,
+        }
+
+        for tc_name, offset in reference_offsets.items():
+            data[tc_name] = data[tc_name] - offset
+
+        return data
+
+    def find_nonstandard_thermocouples(self) -> None:
+        """Make a list of all thermocouples with nonstandard
+        air nozzle configurations in their cells.
+        """
+        nonstandard_thermocouples: List[ThermocoupleData] = []
 
-    Returns
-    -------
-    data : pandas dataframe
-        Time binned dataframe of EFD temperatures where the index
-        is time and the columns are thermocouple temperature with the cold
-        junction temperature removed.
-    """
-    data["coldJunctionMean"] = data[
-        [
-            f"coldJunction{Scanner.TS_01}",
-            f"coldJunction{Scanner.TS_02}",
-            f"coldJunction{Scanner.TS_03}",
-            f"coldJunction{Scanner.TS_04}",
-        ]
-    ].mean(axis=1)
-    for column in data.columns:
         for thermocouple in ThermocoupleTable:
-            if thermocouple.name == column:
-                scanner_number = str(thermocouple.scanner.value)
-                data[column] = np.array(data[column]) - np.array(
-                    data["coldJunction" + scanner_number] - data.coldJunctionMean
-                )
-    return data
+            nozzle_status = [
+                s.nozzle.value
+                for s in AirNozzleTable
+                if s.cell == thermocouple.core_location
+            ][0]
+            if nozzle_status in ["blocked", "sshort", "covered"]:
+                nonstandard_thermocouples.append(thermocouple)
+
+        self.nonstandard_thermocouples = nonstandard_thermocouples
+
+    def coordinate_map(
```

Comment:
The coordinate map is a nice utility, but when you call it outside of the class (e.g. thermocouples.coordinate_map()) it doesn't really return what I expected as a user (and x, y, z mapping for all thermocouples by name). Could we add a quick instance to do that? I know you are using the coordinate map function for lots of different things here, so maybe it would be a new instance with a separate name (e.g. thermocouple_spatial_map).

Reply:
Yeah I think an internal function makes more sense. It's not necessarily wrong to return it, just (as a user) it didn't return what I expected (which was the map) when I called it.